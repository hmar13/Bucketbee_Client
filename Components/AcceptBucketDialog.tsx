import React from 'react';
import theme from '../styles/theme.style';
import Dialog from 'react-native-dialog';

import { View } from 'react-native';

interface AcceptBucketDialogProps {
  acceptDialogVisible: boolean;
  setAcceptDialogVisible(val: boolean): void;
  handleBucketAdd(): void;
}

const AcceptBucketDialog: React.FC<AcceptBucketDialogProps> = ({
  acceptDialogVisible,
  setAcceptDialogVisible,
  handleBucketAdd,
}) => {
  return (
    <View>
      <Dialog.Container visible={acceptDialogVisible}>
        <Dialog.Title>Add this bucket?</Dialog.Title>
        <Dialog.Button
          color="red"
          bold={true}
          label="No"
          onPress={() => setAcceptDialogVisible(false)}
        />
        <Dialog.Button
          color={theme.PRIMARY_COLOR_LITE}
          bold={true}
          label="Yes"
          onPress={handleBucketAdd}
        />
      </Dialog.Container>
    </View>
  );
};

export default AcceptBucketDialog;
