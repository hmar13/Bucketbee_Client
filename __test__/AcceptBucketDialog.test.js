import React from 'react';
import Dialog from 'react-native-dialog';
import renderer from 'react-test-renderer';
import AcceptBucketDialog from '../Components/AcceptBucketDialog';

import { shallow } from 'enzyme';
jest.useFakeTimers();

describe('<AcceptBucketDialog>', () => {
  const onPress = jest.fn()
  const handleBucketAdd = jest.fn()
  const wrapper = shallow(<AcceptBucketDialog acceptDialogVisible={true} setAcceptDialogVisible={onPress} handleBucketAdd={handleBucketAdd} />);

  test('Renders correctly', () => {
    const tree = renderer.create(<AcceptBucketDialog acceptDialogVisible={true} setAcceptDialogVisible={onPress} handleBucketAdd={handleBucketAdd} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('Should render the Title Text Element', () => {
    expect(wrapper.find(Dialog.Button)).toHaveLength(2)
  });
  it('should close dialog when pressed "No" pressed', () => {
    wrapper.find(Dialog.Button).at(0).simulate('press');
    expect(onPress).toHaveBeenCalled();
  })
  it('should add bucket when "Yes" is pressed', () => {
    wrapper.find(Dialog.Button).at(1).simulate('press');
    expect(handleBucketAdd).toHaveBeenCalled();
  })
});
