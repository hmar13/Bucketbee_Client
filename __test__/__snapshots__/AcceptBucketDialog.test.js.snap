// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AcceptBucketDialog> Renders correctly 1`] = `
<View>
  <Modal
    animationType="none"
    hardwareAccelerated={false}
    onHide={[Function]}
    renderToHardwareTextureAndroid={true}
    transparent={true}
    visible={true}
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "black",
          "bottom": 0,
          "left": 0,
          "opacity": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    />
    <View
      needsOffscreenAlphaCompositing={true}
      pointerEvents="box-none"
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
          "opacity": 0,
          "transform": Array [
            Object {
              "scale": 1.2,
            },
          ],
        }
      }
    >
      <View
        onLayout={[Function]}
        style={
          Array [
            Object {
              "marginTop": 22,
            },
            Object {
              "paddingBottom": 0,
            },
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "borderRadius": 13,
                "flexDirection": "column",
                "overflow": "hidden",
                "width": 270,
              },
              Object {},
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "backgroundColor": "rgb(255,255,255)",
                  "bottom": 0,
                  "left": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                },
                Object {},
              ]
            }
          />
          <View
            style={
              Array [
                Object {
                  "margin": 18,
                },
                Object {},
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "black",
                    "fontSize": 18,
                    "fontWeight": "600",
                    "textAlign": "center",
                  },
                  undefined,
                ]
              }
            >
              Add this bucket?
            </Text>
          </View>
          <View
            style={
              Array [
                Object {
                  "borderTopColor": "#A9ADAE",
                  "borderTopWidth": 0.5,
                  "flexDirection": "row",
                  "height": 46,
                  "justifyContent": "space-between",
                },
                Object {},
              ]
            }
          >
            <View
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "flex": 1,
                  "justifyContent": "center",
                  "opacity": 1,
                }
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "backgroundColor": "transparent",
                      "fontSize": 17,
                      "textAlign": "center",
                    },
                    Object {
                      "color": "red",
                      "fontWeight": "600",
                    },
                    undefined,
                  ]
                }
              >
                No
              </Text>
            </View>
            <View
              style={
                Array [
                  Object {
                    "backgroundColor": "#A9ADAE",
                    "height": "100%",
                    "width": 0.5,
                  },
                  Object {},
                ]
              }
            />
            <View
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "flex": 1,
                  "justifyContent": "center",
                  "opacity": 1,
                }
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "backgroundColor": "transparent",
                      "fontSize": 17,
                      "textAlign": "center",
                    },
                    Object {
                      "color": "#18b0b0",
                      "fontWeight": "600",
                    },
                    undefined,
                  ]
                }
              >
                Yes
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </Modal>
</View>
`;
